Guides/Modeling and Binding.md

```diff
@@ -0,0 +1,584 @@
+        self.text = text
+    }
+
+    // MARK: ListDiffable
+
+    func diffIdentifier() -> NSObjectProtocol {
+        return (username + text) as NSObjectProtocol
+    }
+
+    func isEqual(toDiffableObject object: ListDiffable?) -> Bool {
+        return true
+    }
+
+}
+```
+
+A note on the `isEqual(toDiffableObject:)` implementation: Whatever you use to derive the `diffIdentifier` can be omitted from any equality checks, since by definition the objects have already matched on their identifiers.
+
+In this case, the `username` and `text` **must be equal** by the time two objects are checked for equality.
+
+</p></details>
+
+Using the `Comment` array on a `Post` should make some sense: there are a dynamic number of comments on each post. For each comment, you want to display a cell.
+
+What might be a little bit of a new concept, though, is that you need to create models for the `UserCell`, `ImageCell`, _and_ `ActionCell` as well when working with `IGListBindingSectionController`.
+
+> A binding section controller is almost like a mini-`IGListKit`. It takes an array of view models and turns them into configured cells. Get into the habit of creating a new model for each cell type withing an `IGListBindingSectionController` instance.
+
+With that in mind, let's start with the model for the `UserCell`:
+
+Create a new Swift file called **UserViewModel.swift**:
+
+```swift
+import IGListKit
+
+final class UserViewModel: ListDiffable {
+
+  let username: String
+  let timestamp: String
+
+  init(username: String, timestamp: String) {
+    self.username = username
+    self.timestamp = timestamp
+  }
+
+  // MARK: ListDiffable
+
+  func diffIdentifier() -> NSObjectProtocol {
+    // 1
+    return "user" as NSObjectProtocol
+  }
+
+  func isEqual(toDiffableObject object: ListDiffable?) -> Bool {
+    // 2
+    guard let object = object as? UserViewModel else  { return false }
+    return username == object.username
+    && timestamp == object.timestamp
+  }
+
+}
+```
+
+1. Since there will only be **one `UserViewModel` per `Post`**, you can hardcode an identifier. This will enforce only a single model & cell being used.
+2. It's important to write a good equality method for these view models. Anytime something changes, forcing the models to not be equal, the cell will be refreshed.
+
+Try to make view models for the **image** and **action** cell. Remember there is only a single cell per `Post`, so you can use `UserViewModel` as a starting point for how the models should look.
+
+<details>
+  <summary>View model implementations</summary>
+  <p>
+
+```swift
+import IGListKit
+
+final class ImageViewModel: ListDiffable {
+
+  let url: URL
+
+  init(url: URL) {
+    self.url = url
+  }
+
+  // MARK: ListDiffable
+
+  func diffIdentifier() -> NSObjectProtocol {
+    return "image" as NSObjectProtocol
+  }
+
+  func isEqual(toDiffableObject object: ListDiffable?) -> Bool {
+    guard let object = object as? ImageViewModel else { return false }
```

Comment:
Could I propose that you suggest that they `fatalError("expected ImageViewModel but got something else")` (wording?) here? Something is _seriously_ wrong in their setup of they send the wrong ViewModel to a cell.